parsec/core/gui/main_window.py

```diff
@@ -485,7 +485,9 @@ async def _on_finished() -> None:
                 return
             device, auth_method, password = widget.status
             self.reload_login_devices()
-            await self.try_login(device, auth_method, password)
+            # mypy: Cast is safe here because we only need the device slug which
+            # is present in both `LocalDevice` and `AvailableDevice`
+            await self.try_login(cast(LocalDevice, device), auth_method, password)
```

Comment:
Nop, convert the device to an `AvailableDevice` (`get_available_device()`). It's true that we only use the slug right now, but we may use something else in the future and it will be very confusing to the dev who has to do it if the device in device_recovery is sometimes a `LocalDevice`, sometimes an `AvailableDevice`